amazon-ssm-agent: 3.3.2299.0 -> 3.3.2958.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for amazon-ssm-agent is: Agent to enable remote management of your Amazon EC2 instance configuration

meta.homepage for amazon-ssm-agent is: https://github.com/aws/amazon-ssm-agent

meta.changelog for amazon-ssm-agent is: https://github.com/aws/amazon-ssm-agent/releases/tag/3.3.2471.0


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.3.2471.0 with grep in /nix/store/33rfc23ki6b797iis09nnx23hhjm11h1-amazon-ssm-agent-3.3.2471.0
- found 3.3.2471.0 in filename of file in /nix/store/33rfc23ki6b797iis09nnx23hhjm11h1-amazon-ssm-agent-3.3.2471.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

amazon-ssm-agent
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/33rfc23ki6b797iis09nnx23hhjm11h1-amazon-ssm-agent-3.3.2471.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A amazon-ssm-agent https://github.com/r-ryantm/nixpkgs/archive/470fb3e9d0fcba0334616051bd6c7060318ee165.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/470fb3e9d0fcba0334616051bd6c7060318ee165#amazon-ssm-agent
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/33rfc23ki6b797iis09nnx23hhjm11h1-amazon-ssm-agent-3.3.2471.0
ls -la /nix/store/33rfc23ki6b797iis09nnx23hhjm11h1-amazon-ssm-agent-3.3.2471.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>amazon-ssm-agent</li>
  </ul>
</details>

---

###### Maintainer pings

cc @manveru @anthonyroussel @arianvp for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wegank commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 410381`
Commit: `1f456b8ce55501559a3cd6e9fa1e84b2f225e2fc`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>amazon-ssm-agent</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>amazon-ssm-agent</summary>
<pre>Testing red hat
Testing darwin
FAIL
FAIL	github.com/aws/amazon-ssm-agent/agent/updateutil	0.363s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateconstants	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateinfo	0.302s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateinfo/mocks	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updatemanifest	0.259s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updatemanifest/mocks	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/bottlerocket	1.154s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/mocks	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/staticpie	0.976s
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updates3util	3.923s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updates3util/mocks	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/version	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/version/versiongenerator	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/versionutil	0.260s
FAIL
make: *** [makefile:424: --quick-integtest] Error 1</pre>
</details>
</details>

wegank commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 410381`
Commit: `1f456b8ce55501559a3cd6e9fa1e84b2f225e2fc`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>amazon-ssm-agent</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>amazon-ssm-agent</summary>
<pre>Testing red hat
Testing darwin
FAIL
FAIL	github.com/aws/amazon-ssm-agent/agent/updateutil	0.333s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateconstants	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateinfo	0.266s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateinfo/mocks	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updatemanifest	0.259s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updatemanifest/mocks	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/bottlerocket	0.190s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/mocks	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updateprecondition/staticpie	0.274s
ok  	github.com/aws/amazon-ssm-agent/agent/updateutil/updates3util	3.822s
?   	github.com/aws/amazon-ssm-agent/agent/updateutil/updates3util/mocks	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/version	[no test files]
?   	github.com/aws/amazon-ssm-agent/agent/version/versiongenerator	[no test files]
ok  	github.com/aws/amazon-ssm-agent/agent/versionutil	0.201s
FAIL
make: *** [makefile:424: --quick-integtest] Error 1</pre>
</details>
</details>

wegank commented:
@ofborg build amazon-ssm-agent

arianvp commented:
Slipped through the cracks. Trying out locally now and see what I can fix

arianvp commented:
The churn of fixing their broken tests is getting a bit tiresome. Thinking of `doCheck = false`ing this

r-ryantm committed:
amazon-ssm-agent: 3.3.2299.0 -> 3.3.2958.0

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for amazon-ssm-agent is: Agent to enable remote management of your Amazon EC2 instance configuration

meta.homepage for amazon-ssm-agent is: https://github.com/aws/amazon-ssm-agent

meta.changelog for amazon-ssm-agent is: https://github.com/aws/amazon-ssm-agent/releases/tag/3.3.2958.0


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.3.2958.0 with grep in /nix/store/m9132jxa0c275j6b5qwp4dw274apykmv-amazon-ssm-agent-3.3.2958.0
- found 3.3.2958.0 in filename of file in /nix/store/m9132jxa0c275j6b5qwp4dw274apykmv-amazon-ssm-agent-3.3.2958.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
3 total rebuild path(s)

3 package rebuild(s)

First fifty rebuilds by attrpath

amazon-ssm-agent
ssm-agent
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/m9132jxa0c275j6b5qwp4dw274apykmv-amazon-ssm-agent-3.3.2958.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A amazon-ssm-agent https://github.com/r-ryantm/nixpkgs/archive/0ae3bb5071c810164855e9f35fc0a9fe2d249793.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/0ae3bb5071c810164855e9f35fc0a9fe2d249793#amazon-ssm-agent
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/m9132jxa0c275j6b5qwp4dw274apykmv-amazon-ssm-agent-3.3.2958.0
ls -la /nix/store/m9132jxa0c275j6b5qwp4dw274apykmv-amazon-ssm-agent-3.3.2958.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>amazon-ssm-agent</li>
  </ul>
</details>

---

###### Maintainer pings

cc @manveru @anthonyroussel @arianvp for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

arianvp committed:
amazon-ssm-agent: stop running tests

there are many and the codebase is a mess. I don't feel like
dealing with this. If someone feels strongly they can re-enable
but I prefer we ship updates again.

wore maintainers needed.
